apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,26 +141,103 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+let hasSlotMatch = false;
```

Comment:
this is a code smell, no need to have global flag that can leak.

lets create this within the withSlot function and return the outcome as part of the api of the function.

We can unify it to name it in a generic fashion (args related) and also return the other flags we are using 

Example of API proposal:

```ts
const { component, hasArgAsProp, hasArgAsSlot  } = withArgsEnhancer(storyCopy)
```